### ISSUE TITLE
vue/require-default-prop: false positive when pass withDefaults argument by reference

### ISSUE THREAD

**Issue by tilonorrinco** [vuejs/eslint-plugin-vue#2993]
**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version:**  9.39.1
- **eslint-plugin-vue version:** 10.5.1 
- **Vue version:** 3.5.25
- **Node version:** 24.11.1
- **Operating System:** Windows 10 pro 22H2

**Please show your full configuration:**

```json5
import { globalIgnores } from 'eslint/config'
import { defineConfigWithVueTs, vueTsConfigs } from '@vue/eslint-config-typescript'
import pluginVue from 'eslint-plugin-vue'

export default defineConfigWithVueTs(
  {
    name: 'app/files-to-lint',
    files: ['**/*.{ts,mts,tsx,vue}'],
  },

  globalIgnores(['**/dist/**', '**/dist-ssr/**', '**/coverage/**']),

  pluginVue.configs['flat/recommended'],
  vueTsConfigs.recommended,
)
```

**What did you do?**	

```vue
<script setup lang="ts">
defineOptions({ name: 'ExampleComponent' });

interface Props {
  optionalProp?: string;
}

const defaultsObj = {
  optionalProp: 'hello!',
}

withDefaults(defineProps<Props>(), defaultsObj);
</script>
```

**What did you expect to happen?**
No "vue/require-default-prop" error for "optionalProp"

**What actually happened?**

**\src\ExampleComponent.vue
  5:3  warning  Prop 'optionalProp' requires default value to be set  vue/require-default-prop

✖ 1 problem (0 errors, 1 warning)

**Repository to reproduce this issue**

https://github.com/tilonorrinco/bug-vue-require-default-prop

Thanks in advance!

**Comment by ValentinYoushkevich** [comment]
Opened #3114 for this.

The rule only understood an inline object literal as the second argument of
`withDefaults()`, so defaults passed by reference (`defaults`), by spread
(`{ ...defaults }`), from a call, or as `{ ... } as const` were all treated as
"no default given". The PR resolves identifiers and spread sources to the object
they refer to, and stays silent when the defaults cannot be determined
statically (an import, a call expression, a computed key).

One note on the snippet in the report: `@vue/compiler-sfc` rejects it, because
the second argument of `withDefaults()` is hoisted out of `setup()` together
with `defineProps()`:

```
[@vue/compiler-sfc] `defineProps()` in <script setup> cannot reference locally
declared variables because it will be hoisted outside of the setup() function.
```

So the defaults object has to live outside `<script setup>` — in a plain
`<script>` block, or in a separate module:

```vue
<script lang="ts">
export const defaultsObj = { optionalProp: 'hello!' }
</script>

<script setup lang="ts">
interface Props {
  optionalProp?: string;
}
withDefaults(defineProps<Props>(), defaultsObj)
</script>
```

That form compiles down to `mergeDefaults(...)` and is perfectly valid; with the
PR neither form is reported any more.
